--- cpan/Time-HiRes/t/HiRes.t.orig
+++ cpan/Time-HiRes/t/HiRes.t
@@ -116,7 +116,7 @@ if (open(XDEFINE, "xdefine")) {
 # completes fine with (say) 30% slosh, and fail otherwise.  If you do that,
 # consider changing over to test.pl at the same time.
 # --A.D., Nov 27, 2001
-my $limit = 0.25; # 25% is acceptable slosh for testing timers
+my $limit = 0.60; # 25% is acceptable slosh for testing timers
 
 sub skip {
     map { print "ok $_ # skipped\n" } @_;
